src/layout/MainLayout.vue

```diff
@@ -27,59 +29,119 @@ let nextId = 0
 const isEmpty = computed(() => tasks.value.length == 0)
 
 function handleAddTask() {
+    tasks.value.map(task => task.editing = false)
+
     const task = ({
         id: ++nextId,
-        name: `Task ${nextId}`,
-        desc: `This is a description`,
+        name: `Task`,
+        desc: ``,
         priority: 'High',
         done: false,
-        date: new Date().toLocaleDateString(),
+        date: formatDate(new Date()),
+        editing: true
     })
-    tasks.value.push(task)
+    tasks.value.unshift(task)
 
     console.log('new task added:', task)
 }
 
 function handleDoneClicked(id: number) {
     const task = tasks.value.find(task => task.id === id)
+    const realId = tasks.value.length - id
     if (task) {
-        tasks.value.splice(id - 1, 1, { ...task, done: !task.done })
+        tasks.value.splice(realId, 1, { ...task, done: !task.done })
         console.log("Task", id, "done modified")
     }
 }
 
+function handleEditRequested(id: number) {
+    tasks.value.map(task => task.editing = false)
```

Comment:
wouldnt be enough if you just make task.editing true for the task with the same id and task.editing false to the rest?